[19.03] pytest-dependency fails to build

## Issue description

This expression fails to build on 19.03:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/pytest-dependency/default.nix

The problem is that its checkPhase is not correct as the `pytest` command cannot be found. I believe that `pytest` should be in the `checkInputs` for it to succeed.

### Steps to reproduce

Run the below inside `nix repl`

```
pkgs = import (fetchTarball https://github.com/NixOS/nixpkgs-channels/archive/cf3e277dd0bd710af0df667e9364f4bd80c72713.tar.gz) {}
:b pkgs.python3Packages.pytest-dependency
```

The error is:

```
nix-repl> :b pkgs.python3Packages.pytest-dependency 
builder for '/nix/store/jcxfjkd3vazv36jmd2728xp9zr1kvqzs-python3.7-pytest-dependency-0.4.0.drv' failed with exit code 127; last 10 log lines:
  Successfully installed pytest-dependency-0.4.0
  /build/pytest-dependency-0.4.0
  post-installation fixup
  shrinking RPATHs of ELF executables and libraries in /nix/store/35gx0xf8f9j1jiwmmy7ly2hx3gvxzjkq-python3.7-pytest-dependency-0.4.0
  strip is /nix/store/0y7jmqnj48ikjh37n3dl9kqw9hnn68nq-binutils-2.31.1/bin/strip
  stripping (with command strip and flags -S) in /nix/store/35gx0xf8f9j1jiwmmy7ly2hx3gvxzjkq-python3.7-pytest-dependency-0.4.0/lib 
  patching script interpreter paths in /nix/store/35gx0xf8f9j1jiwmmy7ly2hx3gvxzjkq-python3.7-pytest-dependency-0.4.0
  checking for references to /build/ in /nix/store/35gx0xf8f9j1jiwmmy7ly2hx3gvxzjkq-python3.7-pytest-dependency-0.4.0...
  running install tests
  /nix/store/63karsgdg7fm3q0if4zfd7apbd8ac1ci-stdenv-linux/setup: line 1311: pytest: command not found
[0 built (1 failed), 8 copied (8.9 MiB), 1.6 MiB DL]
error: build of '/nix/store/jcxfjkd3vazv36jmd2728xp9zr1kvqzs-python3.7-pytest-dependency-0.4.0.drv' failed
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.31, Formbay ML 2 A, noversion`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `""`
 - nixpkgs: `/etc/nixos/nixpkgs`
```

It can be put in `nativeBuildInputs` as is usual for pytest because we have multiple versions of it within `pythonPackages`. However, the tests fail:
```
============================= test session starts ==============================
platform linux -- Python 3.7.3, pytest-4.2.1, py-1.7.0, pluggy-0.8.1
rootdir: /build/pytest-dependency-0.4.0, inifile:
plugins: dependency-0.4.0
collected 20 items

tests/test_01_marker.py ..                                               [ 10%]
tests/test_02_simple_dependency.py ......                                [ 40%]
tests/test_03_class.py ...                                               [ 55%]
tests/test_03_multiple_dependency.py .                                   [ 60%]
tests/test_03_param.py .                                                 [ 65%]
tests/test_03_runtime.py .                                               [ 70%]
tests/test_03_skipmsgs.py F                                              [ 75%]
tests/test_04_automark.py ...                                            [ 90%]
tests/test_04_ignore_unknown.py ..                                       [100%]

=================================== FAILURES ===================================
_________________________________ test_simple __________________________________

ctestdir = <Testdir local('/build/pytest-of-nixbld/pytest-0/test_simple0')>

    def test_simple(ctestdir):
        """One test fails, other dependent tests are skipped.
        This also includes indirect dependencies.
        """
        ctestdir.makepyfile("""
            import pytest

            @pytest.mark.dependency()
            def test_a():
                pass

            @pytest.mark.dependency()
            def test_b():
                assert False

            @pytest.mark.dependency(depends=["test_b"])
            def test_c():
                pass

            @pytest.mark.dependency(depends=["test_c"])
            def test_d():
                pass
        """)
        result = ctestdir.runpytest("--verbose", "-rs")
        result.assert_outcomes(passed=1, skipped=2, failed=1)
        result.stdout.fnmatch_lines("""
            *::test_a PASSED
            *::test_b FAILED
            *::test_c SKIPPED
            *::test_d SKIPPED
        """)
        result.stdout.fnmatch_lines_random("""
            SKIP * test_c depends on test_b
            SKIP * test_d depends on test_c
>       """)

/build/pytest-dependency-0.4.0/tests/test_03_skipmsgs.py:41:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/nix/store/d5pxi68479pb4lp9hn6znvif78gmp7yy-python3.7-pytest-4.2.1/lib/python3.7/site-packages/_pytest/pytester.py:1282: in fnmatch_lines_random
    self._match_lines_random(lines2, fnmatch)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_pytest.pytester.LineMatcher object at 0x7ffff5a3e5f8>
lines2 = ['SKIP * test_c depends on test_b', 'SKIP * test_d depends on test_c']
match_func = <function fnmatch at 0x7ffff71a7d08>

    def _match_lines_random(self, lines2, match_func):
        """Check lines exist in the output.

        The argument is a list of lines which have to occur in the output, in
        any order.  Each line can contain glob whildcards.

        """
        lines2 = self._getlines(lines2)
        for line in lines2:
            for x in self.lines:
                if line == x or match_func(x, line):
                    self._log("matched: ", repr(line))
                    break
            else:
                self._log("line %r not found in output" % line)
>               raise ValueError(self._log_text)
E               ValueError: nomatch: '*::test_a PASSED'
E                   and: '============================= test session starts =============================='
E                   and: 'platform linux -- Python 3.7.3, pytest-4.2.1, py-1.7.0, pluggy-0.8.1 -- /nix/store/0n8slcq8p5x31kc9hncabsqq9y3fpkzp-python3-3.7.3/bin/python3.7'
E                   and: 'cachedir: .pytest_cache'
E                   and: 'rootdir: /build/pytest-of-nixbld/pytest-0/test_simple0, inifile: pytest.ini'
E                   and: 'plugins: dependency-0.4.0'
E                   and: 'collecting ... collected 4 items'
E                   and: ''
E               fnmatch: '*::test_a PASSED'
E                  with: 'test_simple.py::test_a PASSED'
E               fnmatch: '*::test_b FAILED'
E                  with: 'test_simple.py::test_b FAILED'
E               fnmatch: '*::test_c SKIPPED'
E                  with: 'test_simple.py::test_c SKIPPED'
E               fnmatch: '*::test_d SKIPPED'
E                  with: 'test_simple.py::test_d SKIPPED'
E               line 'SKIP * test_c depends on test_b' not found in output

/nix/store/d5pxi68479pb4lp9hn6znvif78gmp7yy-python3.7-pytest-4.2.1/lib/python3.7/site-packages/_pytest/pytester.py:1308: ValueError
----------------------------- Captured stdout call -----------------------------
============================= test session starts ==============================
platform linux -- Python 3.7.3, pytest-4.2.1, py-1.7.0, pluggy-0.8.1 -- /nix/store/0n8slcq8p5x31kc9hncabsqq9y3fpkzp-python3-3.7.3/bin/python3.7
cachedir: .pytest_cache
rootdir: /build/pytest-of-nixbld/pytest-0/test_simple0, inifile: pytest.ini
plugins: dependency-0.4.0
collecting ... collected 4 items

test_simple.py::test_a PASSED
test_simple.py::test_b FAILED
test_simple.py::test_c SKIPPED
test_simple.py::test_d SKIPPED

=================================== FAILURES ===================================
____________________________________ test_b ____________________________________

    @pytest.mark.dependency()
    def test_b():
>       assert False
E       assert False

test_simple.py:9: AssertionError
=========================== short test summary info ============================
SKIPPED [1] /nix/store/m362v4x3n6004dm2r857jrmhrknjrzym-python3.7-pytest-dependency-0.4.0/lib/python3.7/site-packages/pytest_dependency.py:88: test_c depends on test_b
SKIPPED [1] /nix/store/m362v4x3n6004dm2r857jrmhrknjrzym-python3.7-pytest-dependency-0.4.0/lib/python3.7/site-packages/pytest_dependency.py:88: test_d depends on test_c
================ 1 failed, 1 passed, 2 skipped in 0.02 seconds =================
===================== 1 failed, 19 passed in 1.61 seconds ======================
```

--- a/pkgs/development/python-modules/pytest-dependency/default.nix
+++ b/pkgs/development/python-modules/pytest-dependency/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, buildPythonPackage, fetchPypi, pytest }:
+{ stdenv, buildPythonPackage, fetchPypi, fetchpatch, pytest }:
 
 buildPythonPackage rec {
   version = "0.4.0";
@@ -9,7 +9,18 @@ buildPythonPackage rec {
     sha256 = "bda0ef48e6a44c091399b12ab4a7e580d2dd8294c222b301f88d7d57f47ba142";
   };
 
+  patches = [
+    # Fix tests for pytest>=4.2.0. Remove with the next release
+    (fetchpatch {
+      url = "https://github.com/RKrahl/pytest-dependency/commit/089395bf77e629ee789666361ee12395d840252c.patch";
+      sha256 = "1nkha2gndrr3mx11kx2ipxhphqd6wr25hvkrfwzyrispqfhgl0wm";
+      excludes = [ "doc/src/changelog.rst" ];
+    })
+  ];
+
   propagatedBuildInputs = [ pytest ];
+
+  checkInputs = [ pytest ];
 
   checkPhase = ''
     pytest

